app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-page-views.js

```diff
@@ -74,6 +75,21 @@ window.GOVUK.analyticsGa4.analyticsModules = window.GOVUK.analyticsGa4.analytics
       return window.location.search
     },
 
+    getSearchTerm: function () {
+      var queryString = this.getSearch()
+
+      if (queryString.indexOf('&keywords=') === -1 && queryString.indexOf('?keywords=') === -1) {
+        return undefined
+      }
+
+      var searchTerm = queryString.match(/keywords=([^&]*)/)
```

Comment:
I think some checking could make this code end quicker on non-relevant pages, as well as cut down some of the checks. Could something like this work?

```JavaScript
getSearchTerm: function () {
  var queryString = this.getSearch()

  if (!queryString) {
    return undefined
  }

  var searchTerm = queryString.match(/keywords=([^&]*)/)
  if (!searchTerm) {
    return undefined
  }

  searchTerm = searchTerm[0].replace('keywords=', '')
  searchTerm = this.PIIRemover.stripPIIWithOverride(searchTerm, true, true)
  return searchTerm 
}
```

Reply:
Thanks, will use this instead!